src/antlr/Parser.g

```diff
@@ -1525,8 +1525,8 @@ collectionLiteral returns [Term.Raw value]
 
 listLiteral returns [Term.Raw value]
     @init {List<Term.Raw> l = new ArrayList<Term.Raw>();}
-    @after {$value = new Lists.Literal(l);}
-    : '[' ( t1=term { l.add(t1); } ( ',' tn=term { l.add(tn); } )* )? ']' { $value = new Lists.Literal(l); }
+    @after {$value = new ArrayLiteral(l);}
+    : '[' ( t1=term { l.add(t1); } ( ',' tn=term { l.add(tn); } )* )? ']' { $value = new ArrayLiteral(l); }
```

Comment:
I noticed this previously but isn't this a duplication? We either need `@after {$value = new ArrayLiteral(l);}` or we need `{ $value = new ArrayLiteral(l); }` but I don't think we need both of them.

Reply:
I agree, but parser scares me as we don't have a lot of edge case testing in-tree atm... so I strongly prefer not changing this in this patch